dev/io.openliberty.microprofile.health.4.0.internal/src/io/openliberty/microprofile/health40/internal/HealthFileUtils.java

```diff
@@ -31,7 +31,43 @@ public class HealthFileUtils {
     public static File getHealthDirFile() {
 
         if (healthDirFile == null) {
-            File serverConfigDirFile = new File(System.getProperty("server.output.dir"));
+
+            String serverOutputDir = System.getProperty("server.output.dir").trim();
+
+            File serverConfigDirFile;
+
+            /*
+             * server.output.dir may not be set.
+             * Construct the directory with WLP_OUTPUT_DIR and wlp.server.name or SERVER_NAME
+             */
+            if (serverOutputDir == null || serverOutputDir.isEmpty()) {
+                String wlpOutputDirEnv = System.getenv("WLP_OUTPUT_DIR").trim();
+
+                if (wlpOutputDirEnv == null || wlpOutputDirEnv.isEmpty()) {
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                        Tr.debug(tc,
+                                 "The /health directory can not be created. The server was unable to resolve server.output.dir property nor the WLP_OUTPUT_DIR environment variable.");
+                    }
+                    return null;
+                }
+
+                String serverName = System.getProperty("wlp.server.name").trim();
+
+                if (serverName == null || serverName.isEmpty()) {
+                    serverName = System.getenv("SERVER_NAME").trim();
+                    if (serverName == null || serverName.isEmpty()) {
+                        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                            Tr.debug(tc,
+                                     "The /health directory can not be created. The server was unable to resolve wlp.server.name property nor the SERVER_NAME environment variable.");
```

Comment:
Should this be a WARNING instead, to let users know what went wrong, and why the File Health mechanism didn't get started?

Reply:
Added a message, and modified debug message.